Bottom: a9da469974cfa42ed4d12e87dcf1a11076b4d2e9
Top:    1e1397ae367fe714f62e64d2777ce73da6592706
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-06-08 01:30:11 +0100

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index 1a3d319..15b3534 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -308,6 +308,16 @@ original_poly (PolygonType * p)
       if (n == p->PointN - 1 ||
           (hole < p->HoleIndexN && n == p->HoleIndex[hole] - 1))
         {
+          if (contour == NULL)
+            {
+              printf ("How did that escape - did the loop iterate zero times??\n");
+              POLYGONPOINT_LOOP (p);
+                {
+                  printf ("Hello\n");
+                }
+              END_LOOP;
+              return NULL;
+            }
           poly_PreContour (contour, TRUE);
 
           /* make sure it is a positive contour (outer) or negative (hole) */
